### PR TITLE
Param handling

### PR DESCRIPTION
**Pull request by antdjohns**
Arguments were not being included in request paths.

### CODE REVIEW

File: polygon/rest/base.py [hunk 1]
```diff
@@ -74,11 +74,6 @@ def _get(
         raw: bool = False,
         options: Optional[RequestOptionBuilder] = None,
     ) -> Any:
-        if params is None:
-            params = {}
-        params = {str(k): str(v) for k, v in params.items() if v is not None}
-        logger.debug("_get %s params %s", path, params)
-
```
Comment (antdjohns) [inline polygon/rest/base.py:81]: Fields, which is taking params can be optional; I feel like this is "debug code" that was never removed; same logic is taken care of In the _get_params function. Seems redundant/dead code.

File: polygon/rest/quotes.py [hunk 2]
```diff
@@ -115,7 +115,7 @@ def get_real_time_currency_conversion(
         self,
         from_: str,
         to: str,
-        amount: float,
+        amount: Optional[float] = None,
```
Comment (antdjohns) [inline polygon/rest/quotes.py:118]: inorder for parameters to be used in get_params, they must have set defaults. This change allowed us to remove line 136-139

File: polygon/rest/base.py (hunk 1 above)
Comment (jbonzo) [reply polygon/rest/base.py:81]: I'm not sure if it's debug code. It seems to be converting every item into a string

Comment (jbonzo) [reply polygon/rest/base.py:81]: The tests are fine with this change I'm guessing?

Comment (antdjohns) [reply polygon/rest/base.py:81]: Yes, tests are passing. 
values are added to params only if they are not empty. I can’t think of a reason why we would want to build a new dict based on what was provided and ensure that all keys are type str.

File: polygon/rest/reference.py [hunk 3]
```diff
@@ -238,7 +239,7 @@ def get_ticker_types(
 
         return self._get(
             path=url,
-            params=params,
+            params=self._get_params(fn=self.get_ticker_types, caller_locals=locals()),
```
[the remark this answers is not in the document]
Comment (antdjohns) [reply polygon/rest/reference.py:242]: named params; habit. ill remove.

**Review by jbonzo — APPROVED** [review]

File: polygon/rest/quotes.py (hunk 2 above)
Comment (clickingbuttons) [reply polygon/rest/quotes.py:118]: This param isn't optional in the docs but apparently defaults to 100 :shrug:
image

File: polygon/rest/base.py (hunk 1 above)
Comment (clickingbuttons) [reply polygon/rest/base.py:81]: You're right. This code came before I figured out the introspection for `_get_params`

**Pull request merged**